src/ac_copilot_trainer/modules/realtime_coaching.lua

```
@@ -1,360 +1,350 @@
 -- ---------------------------------------------------------------------------
 
-local NUM_BUCKETS = 1000          -- quantization resolution for O(1) lookup
-local EXIT_WINDOW_SPLINE = 0.008  -- spline distance past corner s1 for "exiting" phase
-local APPROACH_DEFAULT_M = 200    -- fallback approach distance in meters
+local BRAKE_NOW_DIST_M     = 50    -- inside this distance, "BRAKE NOW" if too fast
+local PREPARE_DIST_M       = 100   -- inside this, "PREPARE TO BRAKE" if a bit fast
+local BRAKE_OVER_KMH       = 8     -- "too fast" delta for BRAKE NOW
+local PREPARE_OVER_KMH     = 5     -- "too fast" delta for PREPARE TO BRAKE
+local CORNER_DELTA_KMH     = 8     -- in-corner ±delta for "carry more / ease off"
+local APPROACH_DEFAULT_M   = 200   -- max distance ahead we even look at the next brake
+local DEDUP_HOLD_SEC       = 0.6   -- minimum time before re-firing the same hint kind
```

Luacheck (1.2.0)

[warning] unused variable 'DEDUP_HOLD_SEC'

(W211)